api/v1alpha2/addressset_types.go

```diff
@@ -41,10 +43,13 @@ type AddressSetStatus struct {
 type AddressSet struct {
 	metav1.TypeMeta `json:",inline"`
 	// metadata is the standard object's metadata.
+	// +optional
```

Comment:
metadata is optional? I would think `Name` or `GenerateName` is required in metadata